app/src/features/NavigationBreadcrumbs/NavigationBreadcrumbs.tsx

```diff
@@ -88,56 +103,63 @@ const NavigationBreadcrumbs = (): JSX.Element => {
   const isSource = sourceId && location.pathname.endsWith(sourceId);
 
   return (
-    <Breadcrumbs
-      separator={
-        <Icon
-          iconSize={14}
-          icon={IconNames.CHEVRON_RIGHT}
-          className={classes.icons}
-        />
-      }
-      classes={{ separator: classes.separator }}
-    >
-      <Link
-        variant="h5"
-        color="textSecondary"
-        to="/"
-        className={classes.breadCrumbLinks}
+    <div className={classes.breadcrumbsContainer}>
+      <Breadcrumbs
+        {...props}
+        separator={
+          <Icon
+            iconSize={14}
+            icon={IconNames.CHEVRON_RIGHT}
+            className={classes.icons}
+          />
+        }
+        classes={{ separator: classes.separator }}
       >
-        {t("sources")}
-      </Link>
-      {source && (
-        <Link
-          className={clsx(classes.breadCrumbLinks, {
-            [classes.mainCrumb]: source && !mapping && !isEtl,
-          })}
-          variant="h5"
-          color={!isSource ? "textSecondary" : "textPrimary"}
-          to={`/sources/${source.id}`}
-        >
-          {source.name}
-        </Link>
-      )}
-      {isEtl && (
         <Link
-          className={clsx(classes.breadCrumbLinks, {
-            [classes.mainCrumb]: isEtl,
-          })}
           variant="h5"
-          color={!isEtl ? "textSecondary" : "textPrimary"}
-          to={location.pathname}
+          color="textSecondary"
+          to="/"
+          className={classes.breadCrumbLinks}
         >
-          ETL
+          {t("sources")}
         </Link>
-      )}
-      {source && mapping && (
-        <MappingSelectButton
-          mapping={mapping}
-          source={source}
-          onChange={handleMappingChange}
-        />
-      )}
-    </Breadcrumbs>
+        {source && (
+          <Link
+            className={clsx(classes.breadCrumbLinks, {
+              [classes.mainCrumb]: source && !mapping && !isEtl,
+            })}
+            variant="h5"
+            color={!isSource ? "textSecondary" : "textPrimary"}
+            to={`/sources/${source.id}`}
+          >
+            {source.name}
+          </Link>
+        )}
+        {isEtl && (
+          <Link
+            className={clsx(classes.breadCrumbLinks, {
+              [classes.mainCrumb]: isEtl,
+            })}
+            variant="h5"
+            color={!isEtl ? "textSecondary" : "textPrimary"}
+            to={location.pathname}
+          >
+            ETL
+          </Link>
+        )}
+        {source && mapping && (
+          <div>
+            <MappingSelectButton
+              mapping={mapping}
+              source={source}
+              onChange={handleMappingChange}
```

Comment:
I don't get why we need this `onChange` props. Can't we declare and call `handleMappingChange` directly in  `MappingSelectButton`?